### DISCUSSION TITLE
Touch panel

### DISCUSSION THREAD

**Discussion by traineeuser** [kachurovskiy/nanoels#9] [General]
Thank you for publishing a great project. As a result of this project, I even started to tune my lathe. The lineup is the next Tuda Myler lathe. Arduino mega, hybrid stepper, tft touch screen. I got the project to work on the mega and oled screen but I can't change the program to work on the 3.5 "tft screen. Is this even possible?

**Comment by kachurovskiy** [comment]
Hi and welcome to NanoEls. Which 3.5" TFT screen exactly do you have in mind?

**Reply by traineeuser** [reply]
i have two different ones but i will try this
https://www.aliexpress.com/item/1919710024.html?spm=2114.13010708.0.0.3a994c4dk2uVD2

**Reply by kachurovskiy** [reply]
TFT screens are completely different beasts than OLED displays I have currently supported in the code.

You will need to separately code the UI in a special purpose TFT UI program and then modify the NanoEls code to send/receive data to/from the display when UI needs to change or a button is pressed.

It will require weeks of coding even if you know how to do this. If you have never done this before, it will take months.

I wish I could help you here but it's such a big effort for relatively little gain in user experience that I can't support it at the moment. However all the code is open source and you're welcome to roll your own thing!

**Reply by traineeuser** [reply]
I have never even used an arduino before. Yes it will sometimes be completed at least if I get some help. Thank you for this awesome program.

**Comment by traineeuser** [comment]
Would anyone have any ideas on how to make the bigger screen work without touch functions?

**Reply by kachurovskiy** [reply]
Best option today is LCD2004 as explained in https://github.com/kachurovskiy/nanoels#big-display
